include/deal.II/grid/grid_generator.h

```diff
@@ -1427,9 +1427,10 @@ namespace GridGenerator
    * <code>false</code> all indicators are set to 0 (see
    * @ref GlossColorization "the glossary entry on colorization").
    *
-   * All manifold ids are set to zero, and a SphericalManifold is attached
-   * to the triangulation.
-   *
+   * In 2d, all manifold ids are set to zero, and a SphericalManifold is
+   * attached to the triangulation. In 3d, only the manifold id of the curved
```

Comment:
Why is this different in 2d? I would think there are also flat faces in 2d!

Reply:
Then we may also need to fix quarter_hyper_shell()?https://github.com/dealii/dealii/blob/d7aca55de557876b24d95d89fc02f82da4d8d271/source/grid/grid_generator.cc#L6313

Reply:
I would think so. Can you try to address that as well and see what happens?

Reply:
If we remove the spherical manifold on the flat surface, the mesh quality there gets worse as we refine the mesh, as discussed here https://github.com/dealii/dealii/issues/14176